boilerplate/app/components/Button.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { ComponentType } from "react"
```

Comment:
New Button component reflecting all stylistic changes and adding more customization options.